### ISSUE TITLE
Cached tokens are reused incorrectly

### ISSUE THREAD

**Issue by ph448** [contabo/cntb#15]
While using 2 separate Contabo accounts on the same machine, each with its own API user, I noticed that the cached token from one account is being incorrectly reused when querying the other, even when specifying the `--config` option for the invocation. This results in erroneously manipulating the first account, instead of the second one as intended by the user.

This is because the function to get the token from the token cache https://github.com/contabo/cntb/blob/7c68f3843c3181a2bcaa05d7282dbc5148f8ffe9/oauth2Client/cacheToken.go#L35-L64 seems to only check token expiry and does not invalidate the cache if a different `oauth2-user` is read from either the config file or the command line.

**Comment by reoops** [comment]
Yes, absolutely. We didn't consider this use case. Thank you. We will fix it by invalidating the cache.
